pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -174,6 +173,11 @@ func AddToManager(mgr manager.Manager, r *ReconcileClusterPool, concurrentReconc
 		return err
 	}
 
+	// Watch for changes to ClusterDeploymentCustomizations
+	if err := c.Watch(&source.Kind{Type: &hivev1.ClusterDeploymentCustomization{}}, &handler.EnqueueRequestForObject{}); err != nil {
```

Comment:
I'm sure it would be _possible_ to create a functional test by mocking out the manager, but that would be Hard and I'm not aware of anyone having done it.

You should be able to UT the map func itself though. Use the same client mocks as for the other tests, seed the fake db with CDCs and clusterpools in multiple namespaces, and demonstrate that giving the map func a particular CDC returns (a function that returns) the correct pools.